src/io/hdf_stl.h

```diff
@@ -123,8 +123,6 @@ template<> struct h5data_proxy<std::string>
       }
       ref.resize(dim_out);
       herr_t ret = H5Dread(dataset, datatype, H5S_ALL, H5S_ALL, xfer_plist,&(ref[0]));
-      // Erase trailing null character
-      ref.erase (dim_out-1, 1);
```

Comment:
If the code to remove the last character is left in place, the unit test fails with
<pre>
/codes/qmcpack/github/qmcpack/src/io/tests/test_hdf_archive.cpp:250: FAILED:
  REQUIRE( s == s2 )
with expansion:
  "this is a test" == "this is a tes"
</pre>

Do you know where it was failing? Maybe on a particular machine?